File: adventofcode/cpp/year2019/day21/day21_test.cc
#include "adventofcode/cpp/year2019/day21/day21.h"

#include <fstream>
#include <string>

#include "gtest/gtest.h"

#include "adventofcode/cpp/adventofcode.h"

TEST(Year2019Day21, Part1Actual) {
  std::ifstream input("adventofcode/data/year2019/day21/actual.in");
  std::string output = "19352720";
  adventofcode::answer_t a = day21::part1(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

TEST(Year2019Day21, Part2Actual) {
  std::ifstream input("adventofcode/data/year2019/day21/actual.in");
  std::string output = "1143652885";
  adventofcode::answer_t a = day21::part2(input);
  EXPECT_EQ("", a.error);
  EXPECT_EQ(output, a.answer);
  input.close();
}

File: adventofcode/java/year2016/day21/Day21Test.java
package adventofcode.java.year2016.day21;

import java.io.FileReader;
import java.io.IOException;
import java.io.StringReader;

import org.junit.Test;
import org.junit.Assert;

public class Day21Test {
    @Test
    public void part1Actual() throws IOException {
        try (var input = new FileReader("adventofcode/data/year2016/day21/actual.in")) {
            var output = "bfheacgd";
            var result = Day21.part1(input);
            Assert.assertEquals("no error", "", result.error);
            Assert.assertEquals("correct output", output, result.answer);
        }
    }

     @Test
     public void part2Actual() throws IOException {
         try (var input = new FileReader("adventofcode/data/year2016/day21/actual.in")) {
             var output = "gcehdbfa";
             var result = Day21.part2(input);
             Assert.assertEquals("no error", "", result.error);
             Assert.assertEquals("correct output", output, result.answer);
         }
     }
}
